opnode/cmd/main.go

```diff
@@ -1,31 +1,211 @@
 package main
 
 import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/urfave/cli"
+
 	"github.com/ethereum-optimism/optimistic-specs/opnode/node"
-	"github.com/protolambda/ask"
 )
 
-type MainCmd struct {
+var (
+	Version   = ""
+	GitCommit = ""
+	GitDate   = ""
+)
+
+func main() {
+	// Set up logger with a default INFO level in case we fail to parse flags,
+	// otherwise the final critical log won't show what the parsing error was.
+	log.Root().SetHandler(
```

Comment:
Setting up the default global logger is ok here in the `main`, but please avoid using it as much as possible elsewhere in the implementation. Testing with multiple nodes that all use the same logger gets messy.

Reply:
This is following from the bss (l2output submitter). I think this is fine as pretty much everywhere else it's using the created logger. The other option here is to use golang's log package just for the entrypoint (but that won't respect verbosity flags, but IDK if we respect verbosity flags here anyways).